src/Phulp.php

```diff
@@ -74,23 +74,28 @@ class Phulp
      * @param Source $src
      * @param array $tasks
      */
-    final public static function watch(Source $src, array $tasks)
+    public function watch(Source $src, array $tasks)
     {
-        new Watch($src, $tasks);
+        $phulp = $this;
+        new Watch($src, function () use ($phulp, $tasks) {
+            $phulp->start($tasks);
+        });
     }
 
     /**
      * @param string $path
      *
      * @return PipeInterface
      */
-    final public static function dest($path)
+    public function dest($path)
```

Comment:
I just did - and I was able to really simplify the code. Please have a look at it, and test it.